src/workspace.js

```diff
@@ -657,43 +771,79 @@ module.exports = class Workspace extends Model {
   }
 
   openItem (item, options = {}) {
-    const {pane} = options
+    let {pane, split} = options
 
-    if (item == null) return undefined
-    if (pane.isDestroyed()) return item
+    if (item == null) return Promise.resolve()
+    if (pane != null && pane.isDestroyed()) return Promise.resolve(item)
 
-    if (!options.pending && (pane.getPendingItem() === item)) {
-      pane.clearPendingItem()
-    }
+    const uri = options.uri == null && typeof item.getURI === 'function' ? item.getURI() : options.uri
 
-    const activatePane = options.activatePane != null ? options.activatePane : true
-    const activateItem = options.activateItem != null ? options.activateItem : true
-    this.itemOpened(item)
-    if (activateItem) {
-      pane.activateItem(item, {pending: options.pending})
-    }
-    if (activatePane) {
-      pane.activate()
+    let paneLocation
+    if (pane != null) {
+      paneLocation = this.getPaneLocations().find(location => location.getPanes().includes(pane))
     }
 
-    let initialColumn = 0
-    let initialLine = 0
-    if (!Number.isNaN(options.initialLine)) {
-      initialLine = options.initialLine
-    }
-    if (!Number.isNaN(options.initialColumn)) {
-      initialColumn = options.initialColumn
-    }
-    if ((initialLine >= 0) || (initialColumn >= 0)) {
-      if (typeof item.setCursorBufferPosition === 'function') {
-        item.setCursorBufferPosition([initialLine, initialColumn])
-      }
+    // Determine which location to use, unless a split was provided. In that case, make sure it goes
+    // in the center location (legacy behavior)
+    let location
+    if (paneLocation == null && pane == null && split == null && uri != null) {
+      location = this.previousLocations.load(uri)
     }
 
-    const index = pane.getActiveItemIndex()
-    const uri = options.uri == null && typeof item.getURI === 'function' ? item.getURI() : options.uri
-    this.emitter.emit('did-open', {uri, pane, item, index})
-    return item
+    return Promise.resolve(location)
+      .then(location => {
+        if (paneLocation == null) {
+          if (location == null && typeof item.getDefaultLocation === 'function') {
+            location = item.getDefaultLocation()
+          }
+          paneLocation = this.docks[location] || this.getCenter()
+        }
+      })
+      .then(() => {
+        if (pane != null) return pane
+        pane = paneLocation.getActivePane()
+        switch (split) {
+          case 'left': return pane.findLeftmostSibling()
+          case 'right': return pane.findOrCreateRightmostSibling()
+          case 'up': return pane.findTopmostSibling()
+          case 'down': return pane.findOrCreateBottommostSibling()
+          default: return pane
+        }
```

Comment:
the first one is for finding and the second is for creating and we want to make sure we don't create a new one while we're searching. although looking back now, i think that the split shouldn't even be used in the searching part, should it?